torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -376,6 +377,11 @@ class FullyShardedDataParallel(nn.Module):
             avoid sharding specific parameters when using an
             ``auto_wrap_policy`` or if parameters' sharding is not managed by
             FSDP. (Default: ``None``)
+        device_id (Optional[int]): An `int` describing the CUDA device FSDP
```

Comment:
nit: do we need to stay consistent with DDP? DDP's doc allows both `int` and `torch.device`. 

https://github.com/pytorch/pytorch/blob/6ca8272d46a67b56b1cb65bb551bcbf5709f005b/torch/nn/parallel/distributed.py#L428


BTW, Future's device allows int, str, and device:

https://github.com/pytorch/pytorch/blob/6ca8272d46a67b56b1cb65bb551bcbf5709f005b/torch/futures/__init__.py#L22

Reply:
Yep, it should be int or torch.device. Will fix